dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java

```diff
@@ -409,26 +415,35 @@ private void initMetricsReporter() {
         }
     }
 
-    public boolean isSupportMetrics() {
-        return isClassPresent("io.micrometer.core.instrument.MeterRegistry");
-    }
-
-    public static boolean isSupportPrometheus() {
-        return isClassPresent("io.micrometer.prometheus.PrometheusConfig")
-            && isClassPresent("io.prometheus.client.exporter.BasicAuthHttpConnectionFactory")
-            && isClassPresent("io.prometheus.client.exporter.HttpConnectionFactory")
-            && isClassPresent("io.prometheus.client.exporter.PushGateway");
-    }
-
+    /**
+     * init ObservationRegistry(Micrometer)
+     */
+    private void initObservationRegistry() {
+        if (!ObservationSupportUtil.isSupportObservation()) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Not found micrometer-observation or plz check the version of micrometer-observation version if already introduced, need > 1.10.0");
+            }
+            return;
+        }
+        if (!ObservationSupportUtil.isSupportTracing()) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Not found micrometer-tracing dependency, skip init ObservationRegistry.");
+            }
+            return;
+        }
+        Optional<TracingConfig> configOptional = configManager.getTracing();
+        if (!configOptional.isPresent() || !configOptional.get().getEnabled()) {
+            return;
+        }
 
-    private static boolean isClassPresent(String className) {
-        return ClassUtils.isPresent(className, DefaultApplicationDeployer.class.getClassLoader());
+        DubboObservationRegistry dubboObservationRegistry = new DubboObservationRegistry(applicationModel, configOptional.get());
```

Comment:
So only when all the above three conditions are matched will we initiate DubboObservationRegistry?

Reply:
Yes, due to the Observation.class was proposed after version 1.10 of micrometer, when users use springroot2.x, it defaults to 1.9.x, which means Observation.class is missing. We have made the corresponding judgment and provided users with debug level log prompts.

As for Option<TracingConfig>, this is the switch that determines whether Dubbo itself has enabled the Tracing function